### DISCUSSION TITLE
SUBSCRIPTION_KEY

### DISCUSSION THREAD

**Discussion by Monazo97** [ercot/api-specs#45] [Q&A] [answered]
Hi everyone,

I am having trouble making requests from a Python file. I understand that a SUBSCRIPTION_KEY must be provided in the headers of the request, but I can't find where to get it.

Can anyone help me with this?

Thank you!

**Comment by mackermann-ercot** [comment, marked as answer]
Hi Monazo, 

There is a complete Python example for using both subscription key and authorization token in this discussion:

    https://github.com/ercot/api-specs/discussions/34

And here's the initial guide for registration and retrieving the subscription keys for your account:

    https://developer.ercot.com/applications/pubapi/user-guide/registration-and-authentication/

That should give you a good start! 

Regards,
-mackermann, ERCOT

**Reply by DMJ-CV-91913** [reply]
i dont follow, i provided my api key to different GET request examples found at https://developer.ercot.com/applications/pubapi/user-guide/openapi-documentation/
****only with regards to the testing page of the developers ercot portal****

in these examples, I get a response that I don't have access, though I provided my api key in the authorization section of the webpage.  I am not following why I would need to provide anything other than my api key for making a request as ercot is providing the example and test for various {emilID}.

Is there something I need to do differently for Authorization for testing on this webpage?  

test url in this case { https://api.ercot.com/api/public-reports/NP6-905-CD }

curl -X 'GET' \
  'https://api.ercot.com/api/public-reports/NP6-905-CD' \
  -H 'accept: application/json' \
  -H 'Ocp-Apim-Subscription-Key: 4257ddece1024115887e8df928c4ec71' \
  -H 'Authorization: 4257ddece1024115887e8df928c4ec71'

**Reply by CaffeineLab** [reply]
Here is a simple CURL example to follow: [https://github.com/ercot/api-specs/discussions/66](https://github.com/ercot/api-specs/discussions/66)
